sqldelight-runtime/src/commonMain/kotlin/com/squareup/sqldelight/Transacter.kt

```diff
@@ -69,13 +69,14 @@ abstract class Transacter(private val driver: SqlDriver) {
   /**
    * Starts a [Transaction] and runs [body] in that transaction.
    *
+   * @return The returned value from the [body].
    * @throws IllegalStateException if [noEnclosing] is true and there is already an active
    *   [Transaction] on this thread.
    */
-  fun transaction(
+  fun <R> transaction(
     noEnclosing: Boolean = false,
-    body: Transaction.() -> Unit
-  ) {
+    body: Transaction.() -> R
+  ): R {
```

Comment:
This is a binary-incompatible change. We'll need to use an overload and `@JvmName` to create a separate method in the bytecode. And we need to make sure it doesn't break inference on the consumption side.

Reply:
Ah, right. Because we're changing the return value, I only considered the body's return value. 

I guess we could keep but hide the previous implementation:
```kotlin
  @JvmName("transaction")
  internal fun transactionOld(noEnclosing: Boolean = false, body: Transaction.() -> Unit) =
    transaction(noEnclosing, body)

  @JvmName("transactionWithBody")
  fun <R> transaction(
    noEnclosing: Boolean = false,
    body: Transaction.() -> R
  ): R {
    ...
  }
```

What's your standard for breaking binary compatibility? Only on major version bump?

Reply:
Or we deprecate the old one and rename the new one to `transact` and roll with that.

Reply:
You can make the old one `@Deprecated(level = HIDDEN)` to maintain compatibility.

And yes, only major versions for binary incompatibility. For source incompatibility only (which this also might cause) I don't think we care that much.

Reply:
After talking with @AlecStrong we decided to break binary compatibility as it is likely not going to cause any issues in this case.